contrib/babelfishpg_tds/src/backend/tds/tdstypeio.c

```diff
@@ -1189,6 +1192,7 @@ TdsTypeNumericToDatum(StringInfo buf, int scale)
 		decString++;
 
 	res = TdsSetVarFromStrWrapper(decString);
+	pfree(decStringOrig);
```

Comment:
1. I couldn't add comment above, but we should free memory allocated to zeros on line no. 1153
2. Just to safer side and to avoid double free, use if (decStringOrig) then free

Reply:
I think there is chance that decStringOrig is going to be NULL here